Third-party tests failed on Sat Nov 09 2024

Full history of runs listed here: https://github.com/python/typing_extensions/actions/workflows/third_party.yml

pydantic tests are failing:
```
  pdm install -G testing -G email
  shell: /usr/bin/bash -e {0}
  env:
    PIP_DISABLE_PIP_VERSION_CHECK: 1
    FORCE_COLOR: 1
    pythonLocation: /opt/hostedtoolcache/Python/3.13.0/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.13.0/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.0/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.0/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.0/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.13.0/x64/lib
    LD_PRELOAD: /lib/x86_64-linux-gnu/libgcc_s.so.1
[PdmUsageError]: Requested groups not in lockfile: email,testing
Error: Process completed with exit code 1.
```
Looks related to https://github.com/pydantic/pydantic/pull/10727

--- a/.github/workflows/third_party.yml
+++ b/.github/workflows/third_party.yml
@@ -60,19 +60,18 @@ jobs:
         uses: actions/checkout@v4
         with:
           path: typing-extensions-latest
-      - name: Setup pdm for pydantic tests
-        uses: pdm-project/setup-pdm@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-          allow-python-prereleases: true
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        run: uv python install ${{ matrix.python-version }}
       - name: Add local version of typing_extensions as a dependency
-        run: pdm add ./typing-extensions-latest
+        run: uv add --editable ./typing-extensions-latest
       - name: Install pydantic test dependencies
-        run: pdm install -G testing -G email
+        run: uv sync --group testing --group dev
       - name: List installed dependencies
-        run: pdm list -vv  # pdm equivalent to `pip list`
+        run: uv pip list
       - name: Run pydantic tests
-        run: pdm run pytest
+        run: uv run pytest
 
   typing_inspect:
     name: typing_inspect tests

